zebra-consensus/src/checkpoint.rs

```diff
@@ -212,29 +242,29 @@ where
     ) -> Self {
         // All the initialisers should call this function, so we only have to
         // change fields or default values in one place.
-        let (initial_tip_hash, verifier_progress) = match initial_tip {
-            Some((height, hash)) => {
-                if height >= checkpoint_list.max_height() {
-                    (None, Progress::FinalCheckpoint)
-                } else {
-                    metrics::gauge!("checkpoint.verified.height", height.0 as f64);
-                    metrics::gauge!("checkpoint.processing.next.height", height.0 as f64);
-                    (Some(hash), Progress::InitialTip(height))
-                }
-            }
-            // We start by verifying the genesis block, by itself
-            None => (None, Progress::BeforeGenesis),
-        };
+        let (initial_tip_hash, verifier_progress) =
+            progress_from_tip(&checkpoint_list, initial_tip);
+
+        let (sender, receiver) = mpsc::channel();
         CheckpointVerifier {
             checkpoint_list,
             network,
             initial_tip_hash,
             state_service,
             queued: BTreeMap::new(),
             verifier_progress,
+            reset_receiver: receiver,
+            reset_sender: sender,
         }
     }
 
+    /// Reset the verifier progress back to given tip.
+    fn reset_progress(&mut self, tip: Option<(block::Height, block::Hash)>) {
+        let (initial_tip_hash, verifier_progress) = progress_from_tip(&self.checkpoint_list, tip);
+        self.initial_tip_hash = initial_tip_hash;
+        self.verifier_progress = verifier_progress;
```

Comment:
What happens if multiple blocks are bad?

I'm not sure we need to worry about it, because:
* the checkpoint verifier only commits one block at each height
* the finalized tip never goes backwards
* any finalized height is valid for a reset